homeassistant/components/roborock/config_flow.py

```diff
@@ -0,0 +1,112 @@
+"""Config flow for Roborock."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from roborock.api import RoborockClient
+from roborock.containers import UserData
+from roborock.exceptions import RoborockException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_USERNAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import CONF_BASE_URL, CONF_ENTRY_CODE, CONF_USER_DATA, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class RoborockFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Roborock."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self._username: str | None = None
+        self._errors: dict[str, str] = {}
+        self._client: RoborockClient | None = None
+        self._auth_method: str | None = None
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+        self._errors.clear()
```

Comment:
Now that everything is inlined errors can be a local variable so no need to clear each step.